rocky/tests/reports/test_rpki_report.py

```diff
@@ -0,0 +1,122 @@
+from reports.report_types.rpki_report.report import RPKIReport
+
+
+def test_rpki_report_no_ip(mock_octopoes_api_connector, valid_time, hostname):
+    mock_octopoes_api_connector.oois = {
+        hostname.reference: hostname,
+    }
+    mock_octopoes_api_connector.queries = {
+        "Hostname.<hostname[is ResolvedHostname].address": {
+            hostname.reference: [],
+        },
+    }
+
+    report = RPKIReport(mock_octopoes_api_connector)
+
+    data = report.generate_data(str(hostname.reference), valid_time)
+
+    assert data["rpki_ips"] == {}
+    assert all(
+        v == 0
+        for v in [
+            data["number_of_available"],
+            data["number_of_compliant"],
+            data["number_of_valid"],
+            data["number_of_ips"],
+        ]
+    )
```

Comment:
Writing this as individual asserts is better. It's less lines of code, but more importantly if the test fails you will see why. If you write the assert statement using `all` you get no information on what exactly failed:

```
E       assert False
E        +  where False = all(<generator object test_rpki_report_no_ip.<locals>.<genexpr> at 0x7fcac13b01e0>)
```

```suggestion
    assert data["number_of_available"] == 0
    assert data["number_of_compliant"] == 0
    assert data["number_of_valid"] == 0
    assert data["number_of_ips"] == 0
```

Reply:
Agree with @dekkers Unless you're using `pytest.parametrize`, this would make clear which parametrized value fails in the test, however for this purpose it's just easier to validate them line by line (and they aren't parameters anyway).